### ISSUE TITLE
Add support for new AWS Europe (Milan) region, eu-south-1

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#13063]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

New AWS Europe (Milan) region, `eu-south-1`.

### New or Affected Resource(s)

* Top-level [provider configuration](https://www.terraform.io/docs/providers/aws/index.html)

### Potential Terraform Configuration

```hcl
provider "aws" {
  region = "eu-south-1"
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/04/announcing-the-new-aws-europe-milan-region/).
[Blog post](https://aws.amazon.com/blogs/aws/now-open-aws-europe-milan-region/).

Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/12940

**Comment by ewbankkit** [comment]
Until support is added to the AWS SDK and this is integrated into the Terraform AWS Provider, the provider's automatic region validation must be disabled:

```hcl
provider "aws" {
  region = "eu-south-1"

  skip_region_validation = true
}
```

**Comment by ewbankkit** [comment]
Support added in [AWS SDK v1.30.16](https://github.com/aws/aws-sdk-go/releases/tag/v1.30.16):
* https://github.com/terraform-providers/terraform-provider-aws/pull/12970

**Comment by bflad** [comment]
Support for automatic validation (as well as IDs in various data sources) of this new region has been merged and will release with version 2.60.0 of the Terraform AWS Provider, later this week.

---

Please note that this new region requires [a manual process to enable](https://docs.aws.amazon.com/general/latest/gr/rande-manage.html#rande-manage-enable). Once enabled in the console, it takes a few minutes for everything to work properly.

If the region is not enabled properly, or the enablement process is still in progress, you can receive errors like these:

```console
$ terraform apply

Error: error validating provider credentials: error calling sts:GetCallerIdentity: InvalidClientTokenId: The security token included in the request is invalid.
    status code: 403, request id: 142f947b-b2c3-11e9-9959-c11ab17bcc63

  on main.tf line 1, in provider "aws":
   1: provider "aws" {
```

---

To use this new region before support has been added to Terraform AWS Provider version in use, you must disable the provider's automatic region validation via:

```hcl
provider "aws" {
  # ... potentially other configuration ...

  region                 = "eu-south-1"
  skip_region_validation = true
}
```
